src/sentry/tasks/post_process.py

```diff
@@ -1455,6 +1455,40 @@ def should_postprocess_feedback(job: PostProcessJob) -> bool:
     return False
 
 
+def check_has_high_priority_alerts(job: PostProcessJob) -> None:
+    """
+    Determine if we should fire a task to check if the new issue
+    threshold has been met to enable high priority alerts.
+    """
+    try:
+        event = job["event"]
+        if event.project.flags.has_high_priority_alerts:
+            return
+
+        from sentry.tasks.check_new_issue_threshold_met import (
+            check_new_issue_threshold_met,
+            new_issue_threshold_key,
+        )
+
+        # If the new issue volume has already been checked today, don't recalculate regardless of the value
+        project_key = new_issue_threshold_key(event.project_id)
+        threshold_met = cache.get(project_key)
+        if threshold_met is not None:
+            return
+
+        lock = locks.get(project_key, duration=10)
+        with lock.acquire():
+            check_new_issue_threshold_met.delay(event.project.id)
```

Comment:
You want to do the `cache.set` here, and completely remove all `cache.get`/`cache.set` from the task. That way we know that we only ever fire the task once.

The lock as is doesn't do too much here - `.delay` fires the task straight away, so the task could still be enqueued multiple times by many processes.